src/test/java/de/zalando/aruha/nakadi/partitioning/RandomPartitioningStrategyTest.java

```diff
@@ -0,0 +1,37 @@
+package de.zalando.aruha.nakadi.partitioning;
+
+import com.google.common.collect.ImmutableList;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Set;
+
+import static com.google.common.collect.Sets.newHashSet;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+public class RandomPartitioningStrategyTest {
+
+    private static final RandomPartitioningStrategy strategy = new RandomPartitioningStrategy();
+
+    @Test
+    public void whenManyRunsThenAllPartitionsAreUsed() {
+        final List<String> partitions = ImmutableList.of("a", "b", "c", "d", "e", "f", "g", "h");
+        final Set<String> resolvedPartitions = newHashSet();
+
+        final int numberOfRuns = 1000;
+        for (int run = 0; run < numberOfRuns; run++) {
+            final String resolvedPartition = strategy.calculatePartition(null, null, partitions);
+            resolvedPartitions.add(resolvedPartition);
+        }
+
+        assertThat(resolvedPartitions, equalTo(newHashSet(partitions)));
```

Comment:
I'd say: one shoudl test that: 
a) the data lands in the partition returned by the strategy. this is irrespective of the strategy or how it is implemented. a mock would do the trick and afaik this is covered.
b) for implementations with our logic, test the logic (user defined, hashed) deterministically. for the random case, i suggest something slightly different than what @rcillo suggests about the seed: treat the whole random generation as something you'd mock; say, have a randomNumberProvider that in reality is just Random.nextInt, but under test is mocked to be something you expect it to be, and than the partition computation is deterministic.